### PR TITLE
Merge rebranch into next

### PR DESCRIPTION
**Pull request by bnbarham**
Rebranch is mostly passing now, merge changes into next so we pick up all the fixes.

### CODE REVIEW

**Comment by bnbarham** [comment]
@swift-ci please test

**Review by compnerd — APPROVED** [review]

File: lib/Driver/FrontendUtil.cpp [hunk 1]
```diff
@@ -40,6 +41,22 @@ void swift::driver::ExpandResponseFilesWithRetry(llvm::StringSaver &Saver,
   }
 }
 
+static void removeSupplementaryOutputs(llvm::opt::ArgList &ArgList) {
+  llvm::DenseSet<unsigned> OptSpecifiersToRemove;
+
+  for (llvm::opt::Arg *Arg : ArgList.getArgs()) {
+    if (!Arg)
+      return;
```
[the remark this answers is not in the document]
Comment (compnerd) [reply lib/Driver/FrontendUtil.cpp]: Nevermind, it was fixed in a subsequent commit.

[the remark this answers is not in the document]
Comment (bnbarham) [reply lib/Driver/FrontendUtil.cpp]: Yes, yes it should :). Luckily some tests caught that one.

**Comment by bnbarham** [comment]
I had closed the tab and forgotten about it - if we want to do this then I'll put another one up to avoid conflicts. Otherwise we can just wait until rebranch is merged to main.

**Comment by shahmishal** [comment]
We are close to merging the rebranch into main, so I would say we can hold off. However, I don't have a strong opinion.

**Pull request closed without merging**